src/RandomPlayerbotMgr.cpp

```diff
@@ -2243,6 +2243,15 @@ void RandomPlayerbotMgr::OnBotLoginInternal(Player* const bot)
 {
     LOG_INFO("playerbots", "{}/{} Bot {} logged in", playerBots.size(), sRandomPlayerbotMgr->GetMaxAllowedBotCount(),
              bot->GetName().c_str());
+    
+    if (sPlayerbotAIConfig->randomBotFixedLevel)
+    {
+        bot->SetPlayerFlag(PLAYER_FLAGS_NO_XP_GAIN);
+    }
+    else if (!sPlayerbotAIConfig->randomBotFixedLevel)
```

Comment:
Is there any reason to check the negation here? You're checking the inverse of the bool you just checked, an "else" should cover the same logic without an additional check unless I'm misunderstanding this block

Reply:
Good catch, updated. I had just copied the existing code from PlayerbotMgr.cpp; Lines 552-559